clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -567,6 +568,28 @@ public void testCommitAsyncLeaderEpochUpdate() {
         verify(applicationEventHandler).add(ArgumentMatchers.isA(AsyncCommitEvent.class));
     }
 
+    @Test
+    public void testCommitAsyncPropagatesFencedException() {
```

Comment:
`poll`, `commitSync`, `commitAsync` should all throw FencedIdException.  Can you test each of these API will throw the correct exception after the instance is being fenced?

Reply:
Done